client-core/src/service/sync_state_service.rs

```diff
@@ -182,12 +186,11 @@ mod tests {
             "0F46E113C21F9EACB26D752F9523746CF8D47ECBEA492736D176005911F973A5".to_owned();
         state.trusted_state = trusted_state;
 
-        let key = "Default";
-
-        let storage = MemoryStorage::default();
-        storage.save(KEYSPACE, key, &state).unwrap();
-        let state1: SyncState = storage.load(KEYSPACE, key).unwrap().unwrap();
-
-        assert_eq!(state.encode(), state1.encode());
+        //        let s = r#" {"last_block_height":425,"last_app_hash":"2bc90e1c459ecf0087225490f7d8fa0168e50ecb4d9f48ce4421ebda0dbef273","trusted_state":{"header":{"version":{"block":"10","app":"0"},"chain_id":"test-chain-y3m1e6-AB","height":"425","time":"2020-04-07T10:22:38.614945Z","last_block_id":{"hash":"5D0D5AC9CE9BE2E56242EAAE79CBADEFB1AC02C53BEDEF72E1F36546D38B8E57","parts":{"total":"1","hash":"8FB34ECA32D9D1CC049DEBB83BACD9E91BC13CC51118CC011FC9D45A20272A57"}},"last_commit_hash":"026B4383A67D0C1B3E33FCC828921590737DE4790860B01AB8C3EDAE69C04607","data_hash":null,"validators_hash":"3C21EDBFF3F843947F5DD2C174F5F3621014862CEC172C2731C9439902546E58","next_validators_hash":"3C21EDBFF3F843947F5DD2C174F5F3621014862CEC172C2731C9439902546E58","consensus_hash":"048091BC7DDC283F77BFBF91D73C44DA58C3DF8A9CBC867405D8B7F3DAADA22F","app_hash":"2bc90e1c459ecf0087225490f7d8fa0168e50ecb4d9f48ce4421ebda0dbef273","last_results_hash":null,"evidence_hash":null,"proposer_address":"11D6FD7549C5673EFCE92625FB9D550EC80F40B9"},"validators":{"validators":[{"address":"11D6FD7549C5673EFCE92625FB9D550EC80F40B9","pub_key":{"type":"tendermint/PubKeyEd25519","value":"Nmegn3ZUT0HTHDwqDEujNM7k3C52zD1+YwPp/4khT/c="},"voting_power":"5000194644","proposer_priority":"0"}]}}}"#;
+        //        let state: SyncState = serde_json::from_str(s).unwrap();
+        let s1 = serde_json::to_string_pretty(&state).unwrap();
+        let state2: std::result::Result<SyncState, _> = serde_json::from_str(&s1);
+        let bytes = state.encode();
+        let _state1 = SyncState::decode(&mut bytes.as_slice()).unwrap();
```

Comment:
Where is the assertion? And perhaps give some more meaningful variable names for readability.

Reply:
added one.